pandas/tseries/period.py

```diff
@@ -211,79 +247,16 @@ def _generate_range(cls, start, end, periods, freq, fields):
         return subarr, freq
 
     @classmethod
-    def _from_arraylike(cls, data, freq, tz):
-        if not isinstance(data, (np.ndarray, PeriodIndex,
-                                 DatetimeIndex, Int64Index)):
-            if lib.isscalar(data) or isinstance(data, Period):
-                raise ValueError('PeriodIndex() must be called with a '
-                                 'collection of some kind, %s was passed'
-                                 % repr(data))
+    def _simple_new(cls, data, name=None, freq=None, **kwargs):
```

Comment:
By 'infrastructure' I mean the `Index.__reduce__/__setstate__` methods themselves. 
Or you're saying they're only there to allow back compat? And that we could remove those, default to the normal python machinery, and it would work the same for everything from recent versions?